roles/rsyslog/defaults/main.yaml

```diff
@@ -948,36 +961,51 @@ rsyslog__conf_viaq_elasticsearch:
               } else {
                   set $.viaq_index_name = exec_template("prefix_index_template");
               }
-          {% if openshift_logging_use_ops | default(False) %}
-              if $.viaq_index_prefix startswith "project." then {
+          {% for res in rsyslog_elasticsearch %}
+          {% if loop.length == 1 %}
+              set $.es_host="{{ res.server_host | default('logging-es') }}"
+              set $.es_port="{{ res.server_port | default(9200) }}"
+            {% if use_omelasticsearch_cert | default(False) %}
+              set $.es_ca_cert="{{ res.ca_cert }}"
+              set $.es_cert="{{ res.cert }}"
+              set $.es_key="{{ res.key }}"
+            {% endif %}
+          {% else %}
+          {% if loop.first %}
+              if $.viaq_index_prefix startswith "{{ res.index_prefix }}" then {
+                set $.es_host="{{ res.server_host | default('logging-es') }}"
+                set $.es_port="{{ res.server_port | default(9200) }}"
+          {% if use_omelasticsearch_cert | default(False) %}
+                set $.es_ca_cert="{{ res.ca_cert }}"
+                set $.es_cert="{{ res.cert }}"
+                set $.es_key="{{ res.key }}"
           {% endif %}
+          {% elif loop.last %}
+              else {
```

Comment:
Do we assume here that the last elasticsearch in the `rsyslog_elasticsearch` list is the one all "miscellaneous" logs go to? Why not the first 'Default` es?
If we do assume that, we should mention it in the documentation very clearly...

What will be the index prefix of the logs that go to es-ops ?

We should add the index_prefix parameter to the README.

@nhosoi @richm

Reply:
> Do we assume here that the last elasticsearch in the rsyslog_elasticsearch list is the one all "miscellaneous" logs go to?

Yes

> Why not the first 'Default` es?

We are getting rid of the `default` keyword.

> If we do assume that, we should mention it in the documentation very clearly...

We should document that the elasticsearch outputs will be processed in the order specified, and the last one is the default or "catch-all" output with no condition.

Reply:
> We should document that the elasticsearch outputs will be processed in the order specified, and the last one is the default or "catch-all" output with no condition.

Will do.  Thanks.

Reply:
I'm afraid action is not that dynamic to take variables as a param value... (;_;)
E.g., https://github.com/rsyslog/rsyslog/issues/1149
Note: template seems ok, but not action.   Obviously, the variable param takes a variable, though.
` action(type="mmnormalize" ruleBase="/etc/rsyslog.d/viaq/parse_json.rulebase" variable="$!log")`
I'm putting action in each if/else if clause...

Reply:
@nhosoi not sure what you mean - not sure what you were trying to do

Reply:
This 30-elasticsearch.conf fails with an error [1].  If the serverport param is set to `serverport="9200"`, the error disappears.  First, I thought it could be a type mismatch issue, e.g., string vs integer.  But now I'm suspecting the variable may not be evaluated in the action parameter...  I could be wrong but the issue rsyslog/rsyslog#1149 mentions dynamically updating action may cause thread unsafe situation, thus not supported?>>
```
    set $.es_host = "logging-es";
    set $.es_port = "9200";
    action(
        type="omelasticsearch"
        server="$.es_server"
        serverport="$.es_port"
        template="viaq_template"
        searchIndex="index_template"
        ...
        usehttps="on"
    )
```
[1]
```
Nov 05 18:37:50 ip-172-18-13-50.ec2.internal rsyslogd[24987]: error during parsing file /etc/rsyslog.d/viaq/30-elasticsearch.conf, on or before line 76: parameter 'serverport' is not a proper number [v8.37.0-6.2.el7 try http://www.rsyslog.com/e/2207 ]
Nov 05 18:37:50 ip-172-18-13-50.ec2.internal rsyslogd[24987]: error during parsing file /etc/rsyslog.d/viaq/30-elasticsearch.conf, on or before line 76: errors occured in file '/etc/rsyslog.d/viaq/30-elasticsearch.conf' around line 76 [v8.37.0-6.2.el7 try http://www.rsyslog.com/e/2207 ]
```
What I'm trying to do is very simple - just going back to the previous code, where we don't use the rsyslog local varialbe but put them in action param directly.
```
if $.viaq_index_prefix startswith "project." then {
  action( server="logging-es" serverport="9200" ...)
} else if $.viaq_index_prefix startswith ".operations." then {
  action( server="logging-es-ops" serverport="9200" ...)
} else {
  action( server="logging-es-others" serverport="9200" ...)
}
```

Reply:
Yes - that's the only way you can handle it.